src/steps/generic.rs

```diff
@@ -410,88 +410,82 @@ pub fn run_vcpkg_update(ctx: &ExecutionContext) -> Result<()> {
-/// 2. Just in case, VSCodium could have incompatible changes with VSCode
-pub fn run_vscodium_extensions_update(ctx: &ExecutionContext) -> Result<()> {
-    // Calling vscodoe in WSL may install a server instead of updating extensions (https://github.com/topgrade-rs/topgrade/issues/594#issuecomment-1782157367)
+/// This functions runs for both VSCode and VSCodium, as most of the process is the same for both.
+fn run_vscode_compatible<const VSCODIUM: bool>(ctx: &ExecutionContext) -> Result<()> {
+    // Calling VSCode/VSCodium in WSL may install a server instead of updating extensions (https://github.com/topgrade-rs/topgrade/issues/594#issuecomment-1782157367)
     if is_wsl()? {
         return Err(SkipStep(String::from("Should not run in WSL")).into());
     }
 
-    let vscodium = require("codium")?;
+    let name = if VSCODIUM { "VSCodium" } else { "VSCode" };
+    let bin_name = if VSCODIUM { "codium" } else { "code" };
+    let bin = require(bin_name)?;
 
     // VSCode has update command only since 1.86 version ("january 2024" update), disable the update for prior versions
     // Use command `code --version` which returns 3 lines: version, git commit, instruction set. We parse only the first one
     //
     // This should apply to VSCodium as well.
-    let version: Result<Version> = match Command::new(&vscodium)
+    let version: Result<Version> = match Command::new(&bin)
         .arg("--version")
         .output_checked_utf8()?
         .stdout
         .lines()
         .next()
     {
-        Some(item) => Version::parse(item).map_err(std::convert::Into::into),
-        _ => return Err(SkipStep(String::from("Cannot find vscodium version")).into()),
+        Some(item) => {
+            // Strip leading zeroes because `semver` does not allow them, but VSCodium uses them sometimes.
+            //  This is not the case for VSCode, but just in case, and it can't really cause any issues.
+            let item = item
+                .split('.')
+                .map(|s| s.trim_start_matches('0'))
+                .collect::<Vec<_>>()
+                .join(".");
+            Version::parse(&item).map_err(std::convert::Into::into)
+        }
+        None => return Err(SkipStep(format!("Cannot find {name} version")).into()),
     };
 
-    if !matches!(version, Ok(version) if version >= Version::new(1, 86, 0)) {
-        return Err(SkipStep(String::from(
-            "Too old vscodium version to have update extensions command",
-        ))
-        .into());
-    }
-
-    print_separator("VSCodium extensions");
-
-    ctx.run_type()
-        .execute(vscodium)
-        .arg("--update-extensions")
-        .status_checked()
-}
+    // Raise any errors in parsing the version
+    //  The benefit of handling VSCodium versions so old that the version format is something
+    //  unexpected is outweighed by the benefit of failing fast on new breaking versions
+    let version = version.wrap_err(format!(
```

Comment:
We can use your `output_changed_message!()` macro here, let's me merge this PR first, then you can rebase your `GideonBear:remove-expects` branch to get this updated

Reply:
:+1:

Reply:
Done!